### PR TITLE
Do not allow ebpf proxy mode if konnectivity is disabled

### PR DESCRIPTION
**Pull request by kgroschoff**
### What this PR does / why we need it
Do not allow ebpf proxy mode if konnectivity is disabled

### Which issue(s) this PR fixes

Fixes https://github.com/kubermatic/kubermatic/issues/8496

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by kgroschoff** [comment]
PTAL @maciaszczykm / @floreks

File: src/app/wizard/step/cluster/component.ts [hunk 1]
```diff
@@ -296,10 +305,14 @@ export class ClusterStepComponent extends StepBase implements OnInit, ControlVal
   }
 
   updateCNIPluginOptions() {
-    if (this.controlValue(Controls.CNIPlugin) === CNIPlugin.Cilium) {
-      this.availableProxyModes = ['ipvs', 'iptables', 'ebpf'];
+    if (
+      this.controlValue(Controls.CNIPlugin) === CNIPlugin.Cilium &&
+      !!this.isKonnectivityEnabled &&
```
Comment (floreks) [inline src/app/wizard/step/cluster/component.ts:310]: It's already a boolean. Why the double negation?

Comment (kgroschoff) [reply src/app/wizard/step/cluster/component.ts]: Good catch, it's useless 👍 
Fixed

File: src/app/wizard/step/cluster/component.ts [hunk 2]
```diff
@@ -307,8 +307,8 @@ export class ClusterStepComponent extends StepBase implements OnInit, ControlVal
   updateCNIPluginOptions() {
     if (
       this.controlValue(Controls.CNIPlugin) === CNIPlugin.Cilium &&
-      !!this.isKonnectivityEnabled &&
-      !!this.controlValue(Controls.Konnectivity)
+      this.isKonnectivityEnabled &&
+      this.controlValue(Controls.Konnectivity)
```
Comment (floreks) [inline src/app/wizard/step/cluster/component.ts:311]: Actually, `controlValue` returns `any` and can return `undefined` also. Here, it was ok :slightly_smiling_face:

Comment (kgroschoff) [reply src/app/wizard/step/cluster/component.ts:311]: Good to know. Re-added :D

**Pull request merged**